internal/engine/live_update_build_and_deployer.go

```diff
@@ -47,86 +47,108 @@ func NewLiveUpdateBuildAndDeployer(dcu *containerupdate.DockerContainerUpdater,
 	}
 }
 
+// Info needed to perform a live update
+type liveUpdInfo struct {
+	iTarget      model.ImageTarget
+	state        store.BuildState
+	changedFiles []build.PathMapping
+	runs         []model.Run
+	hotReload    bool
+}
+
 func (lubad *LiveUpdateBuildAndDeployer) BuildAndDeploy(ctx context.Context, st store.RStore, specs []model.TargetSpec, stateSet store.BuildStateSet) (store.BuildResultSet, error) {
 	liveUpdateStateSet, err := extractImageTargetsForLiveUpdates(specs, stateSet)
 	if err != nil {
 		return store.BuildResultSet{}, err
 	}
 
-	if len(liveUpdateStateSet) != 1 {
-		return store.BuildResultSet{}, SilentRedirectToNextBuilderf("LiveUpdateBuildAndDeployer needs exactly one image target (got %d)", len(liveUpdateStateSet))
+	containerUpdater := lubad.containerUpdaterForSpecs(specs)
+	liveUpdInfos := make([]liveUpdInfo, len(liveUpdateStateSet))
+
+	if len(liveUpdateStateSet) == 0 {
+		return nil, RedirectToNextBuilderInfof("no targets for LiveUpdate found")
 	}
 
-	liveUpdateState := liveUpdateStateSet[0]
-	iTarget := liveUpdateState.iTarget
-	state := liveUpdateState.iTargetState
-	filesChanged := liveUpdateState.filesChanged
+	for i, liveUpdateState := range liveUpdateStateSet {
+		iTarget := liveUpdateState.iTarget
```

Comment:
can you break this out into a separate function? all the nesting seems hard to reason about